pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -413,15 +467,25 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 				found.Spec.Template.Spec.Volumes[i].Secret.SecretName = secret.GetName()
 			}
 		}
-		err = r.client.Update(context.TODO(), found)
 
+		err = r.client.Update(context.TODO(), found)
```

Comment:
@rigazilla can we use common updateSecurity function for this?
Maybe we can use common flow for the all Security or Status updates: Update, Get latest CRD, Update whole Status or just Security?

Reply:
@dmvolod in this case we cannot use updateSecurity since we're updating the StatefulSet and only the status for the Infinispan CRD.

Reply:
Sorry, @rigazilla pointing on wrong string. I think about this.
https://github.com/infinispan/infinispan-operator/blob/3eb83d98400b9aa286259ed4816927eb0d597e80/pkg/controller/infinispan/infinispan_controller.go#L413
I not sure and would like to check that the Update without context.Background() cause extra invoke Reconcile

Reply:
I can't see an advantage in wrapping the Update call in another function since we still need the error checking outside to return and requeue from the Reconcile method... but maybe I didn't get your point.

> I not sure and would like to check that the Update without context.Background() cause extra invoke Reconcile

Yeah that's something interesting.
Also a check on how we deal with Update() failure would be useful: the general police is to raise an error and restart the Reconcile loop. This imply that after an error the system is leaved in a clean state and I'm not sure this is true for all our Update() cases.